### ISSUE TITLE
[Feature Request]: Worldborder Visualisation

### ISSUE THREAD

**Issue by Lordimass** [TeamJM/journeymap#790]
### Describe the new feature.

On servers with smaller world borders than default, journeymap continues to map outside of the world border, this means that you end up with a fuzzy border depending on how close you get to the border, making it difficult to tell where the actual border is.

The feature I'm requesting is either a visualisation of the world border on the map (such as a red box) or the ability to hide content that is outside of the world border, so your map could be "complete" by exploring everything in the world border.

I've uploaded a picture of a map for a server I'm working on where my map contains content that is inaccessible to the player, but still shown on the map due to it not hiding outside of the world border. It's scaled down to fit within the filesize limit, but still gets my point across I think.

### Images

![image](https://github.com/user-attachments/assets/e517a3a6-4d64-42bd-870f-cc21ed1fdc0c)

**Comment by mysticdrew** [comment]
How are you feeling defining the world border?

**Comment by Lordimass** [comment]
For me personally I'm just using the vanilla `/worldborder` command, maybe the number could be set in settings for pure client side (default to the default world border), but if the mod is installed server side too then it can communicate the true value to the client?

**Comment by BrainStone** [comment]
Doesn't the server send the worldborder data to the client? I mean how would the client be able to show the worldborder if it didn't knew what it was?

But yeah for starters just the vanilla worldborder would be a great start

**Comment by mysticdrew** [comment]
> Doesn't the server send the worldborder data to the client? I mean how would the client be able to show the worldborder if it didn't knew what it was?

I have no idea honestly, I haven't looked into it yet.

**Comment by MuchPlayer9** [comment]
I feel it would be easy to just have an option in settings were you input the world border size manually and it will hide any map outside the boundary.

**Comment by mysticdrew** [comment]
> I feel it would be easy to just have an option in settings were you input the world border size manually and it will hide any map outside the boundary.

then do it

**Comment by MuchPlayer9** [comment]
> > I feel it would be easy to just have an option in settings were you input the world border size manually and it will hide any map outside the boundary.
> 
> then do it if it is so easy

I have 0 coding knowledge, but I can’t imagine creating code that makes a box the size of the number given, then hiding the area outside of it. Since each block is 1 pixel I’m pretty sure.

**Comment by BrainStone** [comment]
I mean I'd give it a shot if the source was available.

**Comment by gqrshy** [comment]
I too felt this feature was necessary.
If implemented, it would make battleroyale style minigames more interesting.

**Comment by Lordimass** [comment]
Absolutely, it could even be set as a multiplayer admin setting like the ones that already exist.

It also doesn't necessarily need to be a world border, it could just be a limit to the mapping, which might be useful to people in other ways by limiting the radius of map you can have. Maybe it would be useful to map/pack makers?
